util/azure/utils.go

```diff
@@ -204,37 +276,37 @@ func NewDiscoverResourcesSettings(clusterType string, environment string, loginU
    3. creates OperationsMap which is a map of "group": { "resource": { "verb": DataAction{} } } }
    This map is used to create list of AuthorizationActionInfos when we get a SAR request where Resource/Verb/Group is *
 */
-func DiscoverResources(settings *DiscoverResourcesSettings) (OperationsMap, error) {
-	operationsMap := OperationsMap{}
+func DiscoverResources() error {
 	apiResourcesListStart := time.Now()
-	apiResourcesList, err := fetchApiResources(settings)
+	apiResourcesList, err := fetchApiResources()
 	apiResourcesListDuration := time.Since(apiResourcesListStart).Seconds()
 
 	if err != nil {
-		return operationsMap, errors.Wrap(err, "Failed to fetch list of api-resources from apiserver.")
+		return errors.Wrap(err, "Failed to fetch list of api-resources from apiserver.")
 	}
 
 	discoverResourcesApiServerCallDuration.Observe(apiResourcesListDuration)
 
 	getOperationsStart := time.Now()
-	operationsList, err := fetchDataActionsList(settings)
+	operationsList, err := fetchDataActionsList()
 	getOperationsDuration := time.Since(getOperationsStart).Seconds()
 
 	if err != nil {
-		return operationsMap, errors.Wrap(err, "Failed to fetch operations from Azure.")
+		return errors.Wrap(err, "Failed to fetch operations from Azure.")
 	}
 
 	discoverResourcesAzureCallDuration.Observe(getOperationsDuration)
 
-	operationsMap = createOperationsMap(apiResourcesList, operationsList, settings.clusterType)
+	createOperationsMap(apiResourcesList, operationsList)
 
 	klog.V(5).Infof("Operations Map created for resources: %s", operationsMap)
```

Comment:
should this log line be moved to createOperationsMap?

Reply:
no i think this is fine here since it's the final map so i want to print it after function is done

Reply:
While printing, `operationsMap` is accessed without read lock. Hence question was to move it to func where its inside lock.

Reply:
it should be okay to do it without read lock since we are not modifying it at the same time, but i have added it in the other function just to be safe